File: Go/Factory Pattern/simpleShapeFactory.go
package main

type SimpleShapeFactory struct{}

func (s SimpleShapeFactory) CreateShape(shape string) shapeInterface {
	switch shape {
	case "Circle":
		return Circle{}
	case "Rectangle":
		return Rectangle{}
	case "Square":
		return Square{}
	default:
		return nil
	}
}

File: Design Patterns/JS/Factory Pattern/simpleShapeFactory.js
import { ShapeFactory } from "./factory.js";
import { Circle } from "./circle.js";
import { Rectangle } from "./rectangle.js";
import { Square } from "./square.js";

export class SimpleShapeFactory extends ShapeFactory {
  createShape(type) {
    switch (type) {
      case "Circle":
        return new Circle();
      case "Rectangle":
        return new Rectangle();
      case "Square":
        return new Square();
      default:
        return null;
    }
  }
}
